### ISSUE TITLE
orphaned objects are not fully GCed: one refcount key is left in the DB

### ISSUE THREAD

**Issue by warner** [Agoric/agoric-sdk#3378]
## Describe the bug

#3377 was about a kernel crash that occurred when an orphaned object was processed by `processRefcounts()`. The fix avoids the crash, but fails to clean up after the object fully. It leaves the `${kref}.refCount` DB key in place. Normally that would be deleted when a `retireExport` GC action was sent to the exporter, but of course orphaned objects have no exporter. By not pushing a `retireExport` GC action for the late owner vat, we also fail to delete the `.refCount` key.

I'm not sure of the cleanest solution yet. Maybe have a special kind of "no target vat" `retireExport` GC action, which deletes the key but doesn't tell any vat about it? Maybe `processRefcounts()` should notice 1: `recognizable=0` and 2: `ownerVatID === undefined` and delete the key itself, instead of pushing a GC action (but would the DB key change appear in the right commit window?).

The consequence is that we leave one DB key (`${kref}.refCount = 0,0`) around for each orphaned export of any dead vat. We don't kill vats very frequently yet (well, ok, not *intentionally*), so this is low-priority.

**Comment by warner** [comment]
I'm leaning towards the latter fix (have `processRefcounts()` do the deleting immediately.

**Comment by warner** [comment]
Might be associated with (or fixed at the same time as) #3439

**Comment by warner** [comment]
I was working on `test-gc-kernel.js > terminated vat` today, and noticed this clause at the bottom:

https://github.com/Agoric/agoric-sdk/blob/bef63e5f0c6ad6a553b3eef69d722c060995a247/packages/SwingSet/test/test-gc-kernel.js#L1168-L1170

which is commented out because of this bug. I uncommented that part, and it passes. Looking at the code more closely, I think it's been fixed:

https://github.com/Agoric/agoric-sdk/blob/bef63e5f0c6ad6a553b3eef69d722c060995a247/packages/SwingSet/src/kernel/state/kernelKeeper.js#L1276-L1280

That code is run when `processRefcounts` looks at an object with no owner, `reachable === 0`, and `recognizable === 0`, and deletes it. We added this code as part of PR #4957, to fix #4951 (add `syscall.abandonExport`, for liveslots to call during stopVat() back when we did that pre-upgrade), about a year after this ticket was created.

I'll make a PR that uncomments the test clause.
